workerframework-testing-util/src/main/java/com/hpe/caf/worker/testing/QueueManager.java

```diff
@@ -0,0 +1,166 @@
+
+import com.hpe.caf.api.CodecException;
+import com.hpe.caf.api.worker.TaskMessage;
+import com.hpe.caf.util.rabbitmq.DefaultRabbitConsumer;
+import com.hpe.caf.util.rabbitmq.Event;
+import com.hpe.caf.util.rabbitmq.QueueConsumer;
+import com.hpe.caf.util.rabbitmq.RabbitUtil;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.MessageProperties;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Created by ploch on 08/11/2015.
+ */
+public class QueueManager<T> implements Closeable
+{
+    private final QueueServices queueServices;
+    private final WorkerServices workerServices;
+    private String consumerTag;
+
+    private Channel pubChan;
+    private Channel debugPubChan;
+    private Channel conChan;
+    private Channel debugConChan;
+
+    private DefaultRabbitConsumer rabbitConsumer;
+    private Connection connection;
+
+    private String debugInputQueueName;
+    private String debugOutputQueueName;
+    private boolean debugEnabled;
+
+    public QueueManager(QueueServices queueServices, WorkerServices workerServices, boolean debugEnabled)
+    {
+
+        this.queueServices = queueServices;
+        this.workerServices = workerServices;
+        this.debugInputQueueName = this.queueServices.getWorkerInputQueue() + "-debug";
+        this.debugOutputQueueName = this.queueServices.getWorkerResultsQueue() + "-debug";
+        this.debugEnabled = debugEnabled;
+    }
+
+    public Thread start(ResultHandler<T> resultHandler) throws IOException
+    {
+        connection = queueServices.getConnection();
+        pubChan = connection.createChannel();
+        conChan = connection.createChannel();
+        RabbitUtil.declareWorkerQueue(pubChan, queueServices.getWorkerInputQueue(), queueServices.getMaxPriority());
+        if(StringUtils.isNotEmpty(queueServices.getWorkerResultsQueue())) {
+            RabbitUtil.declareWorkerQueue(conChan, queueServices.getWorkerResultsQueue(), queueServices.getMaxPriority());
+        }
+        purgeQueues();
+
+        if (debugEnabled) {
+            debugPubChan = connection.createChannel();
+            debugConChan = connection.createChannel();
+            RabbitUtil.declareWorkerQueue(debugPubChan, debugInputQueueName, queueServices.getMaxPriority());
+            RabbitUtil.declareWorkerQueue(debugConChan, debugOutputQueueName, queueServices.getMaxPriority());
+            debugPubChan.queuePurge(debugInputQueueName);
+            debugConChan.queuePurge(debugOutputQueueName);
+        }
+
+        BlockingQueue<Event<QueueConsumer>> conEvents = new LinkedBlockingQueue<>();
+        SimpleQueueConsumerImpl queueConsumer = new SimpleQueueConsumerImpl(conEvents, conChan, resultHandler, workerServices.getCodec());
```

Comment:
This is one of the objects that you made generic:
```suggestion
        SimpleQueueConsumerImpl<T> queueConsumer = new SimpleQueueConsumerImpl<>(
            conEvents, conChan, resultHandler, workerServices.getCodec());
```